content/docs/reference/buildpack-api.md

```diff
@@ -213,20 +230,55 @@ version = "10.x"
 
 [requires.metadata]
 version-source = ".nvmrc"
+```
 
-[[or]]
+As always, the buildpack `provides` `node`. In this particular case, a version of `node` (`10.x`) is being `requested` in a configuration file (`.nvmrc`), and the buildpack chooses to add an additional piece of metadata (`version-source`), so that it can understand where that request came from.
 
-[[or.provides]]
+#### 3. Two Versions Requested
+The buildpack sees two configuration files (e.g. `.nvmrc` and `buildpack.yml`) that request two different versions of the `node` dependency. One way of dealing with that, would be writing both `requests` to the build plan, and reconciling which version(s) it will provide during the `build` phase. It can provide two different `requires`:
+
+```
+[[provides]]
 name = "node"
 
-[[or.requires]]
+[[requires]]
+name = "node"
+version = "10.x"
+
+[requires.metadata]
+version-source = ".nvmrc"
+
+[[requires]]
 name = "node"
 version = "12.x"
 
-[or.requires.metadata]
+[requires.metadata]
 version-source = "buildpack.yml"
 ```
 
+As always, the buildpack provides `node`. In this case, it requests two different things: `node v10.x` and `node v12.x`. During the `build` phase, it will analyze the two `requires`, and decide which (or both) to follow. Alternatively, it could have resolved the appropriate version in the `detect` phase, and only written the corresponding entry to the build plan.
+
+#### Possible NPM Buildpack
+`NPM`, the default package manager for `node`, is distributed together with node. As a result, a NPM buildpack may require `node`, but not want to `provide` it, trusting that the `node-engine` buildpack will be in charge of `providing` `node`.
+
+It could write the following to the build plan:
+```
+[[requires]]
+name = "node"
+```
+
+If, looking in the `package.json` file, the NPM buildpack see a specific version of `node` requested in the [engines](https://docs.npmjs.com/files/package.json#engines) field (e.g. `14.1`), it may write the following to the build plan:
+```
+[[requires]]
+name = "node"
+version = "14.1"
+
+[requires.metadata]
+version-source = "package.json"
+```
+
+That `requires` will be used by the lifecycle to decide that this (`node-engine` and `npm`) is the correct set of buildpacks, given that the `node-engine` buildpack `provides` `node`, while the `npm` buildpack `requires` `node`, and therefore all `requires` and `provides` are fulfilled.
```

Comment:
Does it not also look at the version being requested? It seems like that information would also be used, no?